lib/sync/index.js

```diff
@@ -17,26 +18,31 @@ function toJSON(dataset_id, returnData, cb) {
   return cb(null, {});
 }
 
+// Invoke the Sync Server.
 function invoke(dataset_id, params, callback) {
   syncUtil.doLog(dataset_id, 'info', 'invoke');
 
+  if (arguments.length < 3) throw new Error('invoke requires 3 arguments');
+
   // Verify that fn param has been passed
   if (!params || !params.fn) {
-    syncUtil.doLog(dataset_id, 'warn', 'no fn parameter provided :: ' + util.inspect(params), params);
-    return callback("no_fn", null);
+    var err = 'no fn parameter provided in params "' + util.inspect(params) + '"';
+    syncUtil.doLog(dataset_id, 'warn', err, params);
+    return callback(err, null);
   }
 
   var fn = params.fn;
 
   // Verify that fn param is valid
-  var fnHandler = invokeFunctions[fn];
+  var fnHandler = module.exports[fn];
   if (!fnHandler) {
-    return callback("unknown_fn : " + fn, null);
+    return callback('invalid fn parameter provided in params "' + fn + '"', null);
   }
 
   return fnHandler(dataset_id, params, callback);
 }
 
+// Initialise cloud data sync service for specified dataset.
 function init(dataset_id, options, cb) {
   syncUtil.doLog(dataset_id, 'info', 'init');
   
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)